samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2418,148 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
+            resources.append(none_datasource)
+
+        for relative_id, function in functions.items():
+            func_config = FunctionConfiguration(
+                logical_id=self.logical_id + relative_id,
+                depends_on=self.depends_on,
+                attributes=self.resource_attributes,
+            )
+
+            # "Id" is a mutually exclusive property to every other property. If this function has it
+            # defined, then we make sure it's the only property, and continue to next function.
+            if function.Id:
+                keys = remove_none_items(function.dict()).keys()  # remove undefined properties, then get keys
+                if len(keys) != 1:
+                    raise InvalidResourceException(
+                        relative_id, "'Id' cannot be defined with other properties in Function."
+                    )
+                continue
+
+            if not function.InlineCode and not function.CodeUri:
+                raise InvalidResourceException(relative_id, "One of 'InlineCode' or 'CodeUri' must be set.")
+
+            if function.InlineCode and function.CodeUri:
+                raise InvalidResourceException(
+                    relative_id, "Both 'InlineCode' and 'CodeUri' cannot be defined at the same time."
+                )
+
+            if function.InlineCode and not function.Runtime:
+                raise InvalidResourceException(
+                    relative_id,
+                    "'Runtime' must be defined when 'InlineCode' is defined.",
+                )
+
+            if not function.DataSource and not function.DataSourceName:
+                raise InvalidResourceException(relative_id, "One of 'DataSource' or 'DataSourceName' must be set.")
+
+            if function.DataSource and function.DataSourceName:
+                raise InvalidResourceException(
+                    relative_id, "Both 'DataSourceType' and 'DataSourceName' cannot be defined at the same time."
+                )
+
+            func_config.Code = function.InlineCode
+            func_config.CodeS3Location = function.CodeUri
+            func_config.ApiId = api_id
+            func_config.Name = function.Name or relative_id
+            func_config.DataSourceName = self._parse_datasource_name(relative_id, function, datasources)
+            func_config.MaxBatchSize = cast(PassThrough, function.MaxBatchSize)
+            func_config.Description = cast(PassThrough, function.Description)
+
+            if function.Sync:
+                func_config.SyncConfig = self._parse_function_sync(function.Sync)
+
+            if function.Runtime:
+                func_config.Runtime = {
+                    "Name": cast(PassThrough, function.Runtime.Name),
+                    "RuntimeVersion": cast(PassThrough, function.Runtime.Version),
+                }
+
+            resources.append(func_config)
+
+        return resources
+
+    def _check_and_construct_none_datasource(
```

Comment:
There are currently a couple of problems with this function (see below) which are hard to capture in translation tests till we have a stable version. Let's add unit tests
- DataSourceName set -> nothing to create
- DataSource set to smth but not None - nothing to create
- DataSource set to NONE, None, none (each of them) and no NoneDataSource exists -> create NoneDataSource
- DataSource set to NONE, None, none (each of them) and NoneDataSource exists -> nothing to create